frontend/azlinux/handle_rpm.go

```diff
@@ -73,13 +73,19 @@ func specToRpmLLB(ctx context.Context, w worker, client gwclient.Client, spec *d
 	specPath := filepath.Join("SPECS", spec.Name, spec.Name+".spec")
 	st := rpm.Build(br, base, specPath, opts...)
 
-	if signer, ok := spec.GetSigner(targetKey); ok {
-		signed, err := frontend.ForwardToSigner(ctx, client, signer, st)
-		if err != nil {
-			return llb.Scratch(), err
-		}
-		st = signed
+	if frontend.SigningDisabled(client) {
+		return st, nil
+	}
+
+	signer := spec.GetSigner(targetKey)
+	if signer == nil {
+		return st, nil
+	}
+
+	signed, err := frontend.ForwardToSigner(ctx, client, signer, st)
+	if err != nil {
+		return llb.Scratch(), err
 	}
 
-	return st, nil
+	return signed, nil
```

Comment:
At this point, should we consider wrapping these steps in a function? So, it can be called in another handler when needed? In this `frontend/windows/handle_zip.go` could be reused?

Reply:
done